microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -223,6 +242,133 @@ def verify_kdumpcrash_large_memory_auto_size(
         self.is_auto = True
         self._kdump_test(node, log_path, log)
 
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. replace FRE bins with CHK bins and reboot the VM
+        2. Configure kdump and reboot VM
+        3. Generate Crash with hvdbg syscall and verify dump
+
+        The test expects the directory containing MSHV CHK binaries tar to be passed
+        in the mshv_chk_bin / mshv_chk_loader testcase variables.
+        """,
+        priority=2,
+    )
+    def verify_mshv_crash(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any],
+        log_path: Path,
+    ) -> None:
+        chkbinpath = variables.get("mshv_chk_bin", "")
+        chkloaderpath = variables.get("mshv_chk_loader", "")
+        if not chkbinpath or not chkloaderpath:
+            raise SkippedException(
+                "Requires a path to MSHV binaries to be passed. "
+                "Please set mshv_chk_bin and mshv_chk_loader testcase variable."
+            )
+        if not isinstance(node.os, CBLMariner):
+            raise SkippedException(
+                f"Testcase only support Mariner-MSHV. Found: {node.os}"
+            )
+
+        mshv = node.tools[Ls].path_exists("/dev/mshv", sudo=True)
+        if not mshv:
+            raise SkippedException("MSHV not present")
```

Comment:
Give more information, how the MSHV is checked.

Reply:
Our kernel will create the /dev/mshv interface. This check is just to make sure that test will run with Mariner only when /dev/mshv (meaning it is running with our kernel) is present.

Reply:
Added description for it now

Reply:
Modified the exception message.